upgrade_downgrade/test_dual_control_plane_upgrade_downgrade_boutique.sh

```diff
@@ -0,0 +1,246 @@
+LOADGEN_NAMESPACE="loadgen"
+FROM_REVISION=$(echo "${FROM_TAG}" | tr '.' '-' | cut -c -20)
+TO_REVISION=$(echo "${TO_TAG}" | tr '.' '-' | cut -c -20)
+
+user="cluster-admin"
+if [[ "${CLOUD}" == "GKE" ]];then
+  user="$(gcloud config get-value core/account)"
+fi
+kubectl create clusterrolebinding cluster-admin-binding --clusterrole=cluster-admin --user="${user}" || echo "clusterrolebinding already created."
+
+write_msg "Reset cluster"
+copy_test_files
+reset_cluster "${TO_ISTIOCTL}"
+
+# Install Initial version of Istio
+write_msg "Deploy Istio ${FROM_TAG}"
+${FROM_ISTIOCTL} install -f "${TMP_DIR}/iop-control-plane.yaml" -y --revision "${FROM_REVISION}" || die "control plane installation failed"
+${FROM_ISTIOCTL} install -f "${TMP_DIR}/iop-gateways.yaml" -y --revision "${FROM_REVISION}" || die "gateway installation failed"
+kubectl wait --for=condition=ready pod --all -n "${ISTIO_NAMESPACE}" --timeout=30m
+
+# Create namespace and label for automatic injection
+kubectl get namespace "${TEST_NAMESPACE}" || kubectl create namespace "${TEST_NAMESPACE}"
+kubectl get namespace "${LOADGEN_NAMESPACE}" || kubectl create namespace "${LOADGEN_NAMESPACE}"
+kubectl label namespace "${TEST_NAMESPACE}" istio-injection- || echo "istio-injection label removed"
+kubectl label namespace "${TEST_NAMESPACE}" istio.io/rev="${FROM_REVISION}"
+
+function deploy_and_wait_for_services() {
+  local -n services="$1"
+  for svc in "${services[@]}"; do
+    SERVICE_MANIFEST_FILE="${TMP_DIR}/boutique/k8s-${svc}.yaml"
+    if [[ ! -f "${SERVICE_MANIFEST_FILE}" ]]; then
+      echo "Failed to deploy: Kubernetes manifest file for ${svc} not found - ${SERVICE_MANIFEST_FILE}"
+      return 1
+    fi
+    kubectl apply -f "${SERVICE_MANIFEST_FILE}" # Namespace is already defined in manifest
+  done
+  kubectl wait --for=condition=ready pod --all -n "${TEST_NAMESPACE}" --timeout=30m
+}
+
+# Setup boutique shop app. Namespace is specified in YAML.
+# Link to architecture diagram:
+# https://github.com/GoogleCloudPlatform/microservices-demo/blob/master/docs/img/architecture-diagram.png 
+base_services=( "redis-cart" \
+                "productcatalogservice" \
+                "currencyservice" \
+                "shippingservice" \
+                "paymentservice" \
+                "emailservice" \
+                "adservice" )
+
+services=( "cartservice" \
+           "checkoutservice" \
+           "recommendationservice" \
+           "frontend" )
+
+if ! deploy_and_wait_for_services base_services; then exit 1; fi
+if ! deploy_and_wait_for_services services; then exit 1; fi
+
+kubectl apply -f "${TMP_DIR}/boutique/istio-manifests.yaml"
+kubectl apply -f "${TMP_DIR}/boutique/k8s-loadgenerator.yaml"
+kubectl wait --for=condition=ready pod -n "${LOADGEN_NAMESPACE}" --timeout=30m
+
+# Start external traffic from fortio
+# 1. First get ingress address
+# 2. Next, use that address to fire requests at boutique shop app
+wait_for_ingress
+
+export TRAFFIC_RUNTIME_SEC=800
```

Comment:
better to put the constants at the top of the file.

Reply:
Done.